openshift/dynamic/exceptions.py

```diff
@@ -49,6 +50,17 @@ def __str__(self):
 
         return '\n'.join(error_message)
 
+    def summary(self):
+        if self.body:
+            if self.headers and self.headers.get('Content-Type') == 'application/json':
+                message = json.loads(self.body).get('message')
+                if message:
+                    return message
+
+            return self.body
```

Comment:
I should probably know this, but does body/message generally have the status code in it?

Reply:
An example:
```
(Pdb) type(exc)
<class 'openshift.dynamic.exceptions.ForbiddenError'>
(Pdb) json.loads(exc.body)
{
  'kind': 'Status',
  'apiVersion': 'v1',
  'metadata': {},
  'status': 'Failure',
  'message': 'namespaces is forbidden: User "pilou" cannot list namespaces at the cluster scope: User "pilou" cannot list all namespaces in the cluster: clusterrole.rbac.authorization.k8s.io "access-asb-role" not found',
  'reason': 'Forbidden',
  'details': {'kind': 'namespaces'},
  'code': 403
}
```
It seems related to this code: https://github.com/openshift/origin/blob/6b2bfb68f8f3e19bef24dbaca342567fc4106633/vendor/k8s.io/kubernetes/staging/src/k8s.io/apimachinery/pkg/api/errors/errors.go#L37 and https://github.com/openshift/origin/blob/master/vendor/k8s.io/kubernetes/staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/generated.proto#L687

Is that ok ?